setup.py

```diff
@@ -37,6 +37,7 @@
     "scipy>=1.0",
     "sympy>=1.3",
     "dill>=0.3",
+    "z3-solver>=4.7"
```

Comment:
This should be an option dependency, can we move this to the `extras_require` dict below at L103? Probably add a new entry like:

`'crosstalk-pass': ['z3-solver>=4.7'],`

Reply:
Done.